scipy/interpolate/_cubic.py

```diff
@@ -13,34 +15,36 @@
            "Akima1DInterpolator", "CubicSpline"]
 
 
-def prepare_input(x, y, axis, dydx=None):
+def prepare_input(x, y, axis, dydx=None, xp=None):
     """Prepare input for cubic spline interpolators.
 
     All data are converted to numpy arrays and checked for correctness.
     Axes equal to `axis` of arrays `y` and `dydx` are moved to be the 0th
     axis. The value of `axis` is converted to lie in
     [0, number of dimensions of `y`).
     """
+    if xp is None:
+        xp = array_namespace(x, y, dydx)
```

Comment:
I would prefer if we didn't do this as it would help root out any instances we don't pass `xp`
```suggestion
```